modin/engines/base/frame/data.py

```diff
@@ -546,31 +546,46 @@ def from_labels(self) -> "BasePandasFrame":
             A new BasePandasFrame.
         """
         new_row_labels = pandas.RangeIndex(len(self.index))
+
         # Column labels are different for multilevel index.
-        if len(self.index.names) > 1:
-            # We will also use the `new_column_names` in the calculation of the internal metadata, so this is a
-            # lightweight way of ensuring the metadata matches.
-            new_column_names = pandas.Index(
-                [
-                    self.index.names[i]
-                    if self.index.names[i] is not None
-                    else "level_{}".format(i)
-                    for i in range(len(self.index.names))
-                ]
-            )
-            new_columns = new_column_names.append(self.columns)
+        # We will also use the `new_column_names` in the calculation of the internal metadata, so this is a
```

Comment:
`new_column_names` is no longer assigned here. Please update.

Reply:
Ok.